### DISCUSSION TITLE
github flavored markdown

### DISCUSSION THREAD

**Discussion by bytearchive** [Textualize/rich#1278] [Q&A]
can `rich` output GitHub flavored markdown? or can it be customized?

**Comment by willmcgugan** [comment]
Rich uses commonmark. If there is a GH markdown parser for Python then you could certainly use Rich as a renderer.

**Reply by emeryberger** [reply]
Is it possible to have Rich output Markdown (Commonmark is fine)?

**Reply by willmcgugan** [reply]
```python
from rich.markdown import Markdown
from rich import print
print(Markdown("## It *sure* is!"))
```

**Reply by emeryberger** [reply]
I actually mean the reverse :). -- i.e., produce Markdown as output (but Commonmark is pretty thin; GFM supports tables and the like, making it a potentially reasonable output format for rich).

**Reply by emeryberger** [reply]
(I _suspect_ that this was the intent of @bytearchive's question)

**Reply by willmcgugan** [reply]
Ah, gotcha.

Possible to some degree, but I don't think it could make a good job of it. Once rendered, Rich doesn't have any semantic information re the output, just a grid of characters.

**Comment by tompropst** [comment]
If anyone lands here looking primarily for table support...

```sh
echo "rich >=13" > overrides.txt
uv tool install rich-cli --overrides overrides.txt
```

Since `rich-cli` [currently requires `rich <13.0.0`](https://github.com/Textualize/rich-cli/blob/309b64b498eca1fa764991e9aa3f7a831a9b1b26/pyproject.toml#L25), this may have problems but so far it works well for me.
